cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -112,20 +112,11 @@ protected int getInstances(HttpRequest request) throws BadRequestException {
   }
 
   protected Map<String, String> decodeArguments(HttpRequest request) throws JsonSyntaxException {
-    ChannelBuffer content = request.getContent();
-    if (!content.readable()) {
+    Map<String, String> args = parseBody(request, STRING_MAP_TYPE);
```

Comment:
Actually, nvm. that seems like just a log message, not an error message propagated to users. Where ``parseBody`` catches a runtime exception on line 107 above, should we throw throw a ``BadRequestException`` with a better error message instead? Seems like a user error, so shouldn't respond with 500?